openshift-kube-apiserver/admission/customresourcevalidation/route/validation_opts.go

```diff
@@ -0,0 +1,35 @@
+package route
+
+import (
+	"k8s.io/apiserver/pkg/util/feature"
+	"k8s.io/component-base/featuregate"
+
+	configv1 "github.com/openshift/api/config/v1"
+	routecommon "github.com/openshift/library-go/pkg/route"
+)
+
+type RouteValidationOptionGetter interface {
+	GetValidationOptions() routecommon.RouteValidationOptions
+}
+
+type RouteValidationOpts struct {
+	opts routecommon.RouteValidationOptions
+}
+
+var _ RouteValidationOptionGetter = &RouteValidationOpts{}
+
+func NewRouteValidationOpts() *RouteValidationOpts {
+	return &RouteValidationOpts{}
+}
+
+func (o *RouteValidationOpts) GetValidationOptions() routecommon.RouteValidationOptions {
+	return o.opts
+}
+
+func (o *RouteValidationOpts) Initialize() *RouteValidationOpts {
+	o.opts = routecommon.RouteValidationOptions{
+		AllowExternalCertificates: feature.DefaultMutableFeatureGate.Enabled(featuregate.Feature(configv1.FeatureGateRouteExternalCertificate)),
+	}
+
+	return o
+}
```

Comment:
It isn't clear why you need this whole `RouteValidationOpts` abstraction, as opposed to using `routecommon.RouteValidationOptions` in `validateCustomResourceWithClient`.  Is this `Initialize` method the reason why you need `RouteValidationOpts`?

Reply:
yes, that's correct. Our intent here is to particularly set that validation option upon `Initialize()` only when the tech preview `RouteExternalCertificate` feature gate is enabled.